atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/CellsSweeperShould.java

```diff
@@ -125,14 +126,13 @@ public void releaseTheBackupLockAfterDeleteAndAddingSentinels() {
     }
 
     @Test
-    public void releaseTheBackupLockIfDeleteFails() throws Exception {
-        doThrow(Exception.class).when(mockKvs).delete(TABLE_REFERENCE, SINGLE_CELL_TS_PAIR);
-
-        try {
-            sweeper.sweepCells(TABLE_REFERENCE, SINGLE_CELL_TS_PAIR, ImmutableSet.of());
-        } catch (Exception e) {
-            // expected
-        }
+    public void releaseTheBackupLockIfDeleteFails() {
+        doThrow(new RuntimeException("something bad happened"))
+                .when(mockKvs).delete(TABLE_REFERENCE, SINGLE_CELL_TS_PAIR);
+
+        assertThatExceptionOfType(RuntimeException.class)
+                .isThrownBy(() -> sweeper.sweepCells(TABLE_REFERENCE, SINGLE_CELL_TS_PAIR, ImmutableSet.of()))
+                .withMessage("something bad happened");
```

Comment:
This is a direct translation of the previous test. Lines 129-130 aren't valid in Mockito 2 since `KeyValueService.delete` does *not* throw any checked exceptions. That said, I have to change the assertion, and I'm switching to using the assertJ style.

You're right that the exception doesn't matter, which is why I've picked a `RuntimeException`. But I want to know that I'm not throwing elsewhere from where I'm doing the `doThrow` hence why I'm asserting on the message => I know that a bad thing happening in delete still doesn't matter.

Reply:
Jeremy's concern might have been that the exception gets wrapped at some point, although it looks like that's not the case here.

Reply:
Exactly - my concern was if the exception gets wrapped failing seems harsh.